### PR TITLE
Resolved reformat of blog posts

### PR DESCRIPTION
**Pull request by phillipj**
Hijacked PR #154 as it cannot be merged due to merge conflict issues and lack of response from the original authors.

This also needed cheerio added to package.json, which made the content generation to break. Not trying to make someone look bad, have forgot the same thing myself hundreds of times, but it highlights the need for smoketesting these PRs #225. That alert could have come from Travis when the initial PR was created, and of course prevented a bad merge & deploy.
#### UPDATED AFTER DISCUSSIONS BELOW

The original PR displayed title and first part of the blog post contents for _all_ blog posts ever written. This PR adds pagination by year, and title + blog contents for the first 10 posts, followed by the remaining posts in that year by title only.

Screenshots showing (1) the first page of blog posts as of 2016, clicking "older" will take you to (2) all the blog posts of 2015 and (3) how blog posts look from the 10th posts and below.
##### 1)

screen shot 2016-01-01 at 22 40 48
##### 2)

screen shot 2016-01-01 at 22 40 59
##### 3)

screen shot 2016-01-01 at 22 41 12

### CODE REVIEW

**Comment by phillipj** [comment]
Btw the page of blog posts with these changes gets **crazy** long. Adding some kind of pagination later would be great!

**Comment by fhemberger** [comment]
@phillipj Here is a pagination plugin for metalsmith: https://github.com/RobinThrift/metalsmith-paginate.

IMHO it would be nice to split blog posts into two groups:
- Posts of the current year, using the long format proposed by this PR
- Older posts, using the simple listing we had so far

both paginated. What do you think?

**Comment by phillipj** [comment]
Nice, I'll have a look at that plugin @fhemberger :+1:

**Comment by mrkiffie** [comment]
I noticed that the summaries also contain the anchor links form the titles in the blog posts which I'm not sure is desirable. In addition, some of the posts have a significant document structure which doesn't really translate so well to a summary. 

Here is a screenshot for reference:
blog post summary screenshot

I'm not sure if stripping the tags would suffice as a complete solution, but it could potentially clean up the presentation. 

Perhaps an alternate implementation would be to use cheerio to specifically extract paragraphs?

**Comment by phillipj** [comment]
Sorry for not getting any progress on pagination! I've tried different approaches with `metalsmith-paginate`, but can't get it working properly. Since I'm no metalsmith expert, time flies by while trying to wrap my head around what's happening and debug :(

File: scripts/helpers/summary.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * how the metalsmith-paginate plugin operates,
+ * this helper has to handle two different types of
+ * HTML contents:
+ *  - clean blog posts converted from markdown to HTML,
+ *    seen on the first page of blog posts
+ *  - the remaining paginated pages has gone
+ *    through the handlebars process and therefore has the
+ *    entire page layout (w/<html>, <head> and <body> etc)
+ *    wrapped around the actual blog contents :(
+ */
```
Comment (phillipj) [inline scripts/helpers/summary.js:19]: This sums up my hours of trial'n'error to get this working.

**Comment by phillipj** [comment]
Soooo... Basic pagination works now. It also exludes some specific tags to clean up the summary presentation as suggested by @mrkiffie.

skjermbilde 2015-11-18 kl 11 17 42

**Comment by phillipj** [comment]
> IMHO it would be nice to split blog posts into two groups

@fhemberger sounds like a very good idea! Would you consider it critical for the first release of this redesign or could post grouping be added later?

**Comment by fhemberger** [comment]
I think it might be easier to start with a pagination by post year. Afterwards, try to get a different format for the current year. I have to look into this first, haven't had much time lately to look at the website issues here.

**Comment by phillipj** [comment]
I assume you mean easier for the end-user? As technically I'm guessing we
would have to create a custom metalsmith paginator plugin for that, which
doesn't sound simpler.

**Comment by phillipj** [comment]
Alright, I've been playing with this again. Started working on a metalsmith plugin which paginates by year. After seeing how many blog posts we have per year, it quickly became apparent we cannot show  summary for all.

Some quick stats:

```
2011 had 48 blog posts
2012 had 59 blog posts
2013 had 55 blog posts
2014 had 25 blog posts
2015 had 76 blog posts
```

Currently working on display summary for the latest 10 blog posts per year, but only titles for the following posts. Hopefully be up-n-running by the end of this week.

**Comment by phillipj** [comment]
All finished and ready to go. Ended up creating https://github.com/phillipj/metalsmith-yearly-pagination to do most of the grunt work.

PTAL

**Comment by phillipj** [comment]
ping @fhemberger / anyone, got any thoughts on this? Just updated the PR description with up-to-date info and three screenshots.

I'd like to either merge or close this as it has been hanging around for quite some time now.

**Comment by ghost** [comment]
@phillipj can you remove the standard list style formatting? i think it looks too 'vanilla html'

**Comment by phillipj** [comment]
@sup sure, just pushed the changes and updated the screenshots

**Comment by ghost** [comment]
yup, absolutely looks good to me! this has been mocking me for a longer time now, didn't know there was a PR for it. but from my side, we can merge

**Comment by fhemberger** [comment]
@phillipj Looking at the screenshots and briefly scanning the code, it seems to be a neat solution. Sorry, I don't have much time for a deeper review ATM. So if no one objects, go for it.

**Comment by phillipj** [comment]
Thanks for reviewing!

**Pull request merged**